addons/account_edi/models/account_edi_document.py

```diff
@@ -182,20 +182,14 @@ def _postprocess_cancel_edi_results(documents, edi_result):
                 edi_result = edi_format._cancel_payment_edi(documents.move_id, test_mode=test_mode)
                 _postprocess_cancel_edi_results(documents, edi_result)
 
-    def _process_documents_no_web_services(self):
-        """ Post and cancel all the documents that don't need a web service.
-        """
-        jobs = self.filtered(lambda d: not d.edi_format_id._needs_web_services())._prepare_jobs()
-        for documents, doc_type in jobs:
-            self._process_job(documents, doc_type)
-
-    def _process_documents_web_services(self, job_count=None, with_commit=True):
-        """ Post and cancel all the documents that need a web service. This is called by CRON.
+    @api.model
+    def _process_jobs(self, jobs, with_commit=True):
```

Comment:
I liked `_process_documents_web_services` better since it is a function only useful for web services

Reply:
That method still exist and have a different responsability now :
- `_process_documents_web_services` process all the documents in self.
- `_cron_process_documents_web_services` process a subset of jobs from the documents in self and re-trigger the cron if the subset is smaller than all the jobs.
- `_process_jobs` (called by the two previous methods): deals with transactions and locks on all the jobs passed as argument, independently of whether it uses web services or not. I agree that this function is only used on the documents using webservices but it's not relevant here and not mandatory.
- `_process_job` (called by `_process_jobs`) : the actual processing of one single job.

Reply:
If you have `_process_documents_web_services(self, job_count=None, with_commit=True)`

Then you can have
```python
    @api.model
    def _cron_process_documents_web_services(self, job_count=None):
        ''' Method called by the EDI cron processing all web-services.
        :param job_count: Limit explicitely the number of web service calls. If not provided, process all.
        '''
        edi_documents = self.search([('state', 'in', ('to_send', 'to_cancel'))])
        edi_documents._process_documents_web_services(job_count, with_commit=True)

        # Mark the CRON to be triggered again asap since there is some remaining jobs to process.
        if len(all_jobs) - len(jobs_to_process):
            self.env.ref('account_edi.ir_cron_edi_network')._trigger()
```

It is more DRY, the function name tells what it is supposed to do (`web_service` in the name), you remove one useless function, and you have less diff.